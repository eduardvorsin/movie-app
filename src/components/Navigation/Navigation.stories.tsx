import type { Meta, StoryObj } from '@storybook/react';
import Navigation, { Props } from './Navigation';
import { useTranslation } from 'react-i18next';

const meta: Meta<typeof Navigation> = {
	title: 'Layout/Navigation',
	tags: ['autodocs'],
	argTypes: {
		className: {
			description: 'Additional classes for the button'
		},
		testId: {
			description: 'Id for testing the component',
		},
		dictionary: {
			control: false,
			description: 'An object with keys for localization of the component',
		},
	},
	component: Navigation,
	parameters: {
		docs: {
			description: {
				component: 'The Navigation component in the documentation is a key tool for managing user navigation in the application. It provides transparent and convenient access to various sections and functions by providing intuitive controls such as menus, links and buttons.',
			},
		},
	},
}

export default meta;
type Story = StoryObj<typeof Navigation>;

const NavigationWithHooks = (props: Omit<Props, 'dictionary'>) => {
	const { t } = useTranslation('common');
	const dictionary = {
		movies: t('navigation.movies'),
		persons: t('navigation.persons'),
		tv: t('navigation.tv'),
		new: t('navigation.new'),
		collections: t('navigation.collections'),
	};

	return (
		<Navigation
			{...props}
			dictionary={dictionary}
		/>);
}

export const Default: Story = {
	parameters: {
		docs: {
			description: {
				story: 'The default state of the component',
			},
		},
	},
	render: (args) => <NavigationWithHooks {...args} />
};
