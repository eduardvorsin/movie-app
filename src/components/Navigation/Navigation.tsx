'use client';

import { NavigationRoutes, navigationRoutes } from 'src/constants';
import Link from '@/components/Link/Link';
import { useSelectedLayoutSegments } from 'next/navigation';
import { MouseEventHandler } from 'react';
import { GeneralProps } from '@/types/shared';

export type Props = {
	onClick: MouseEventHandler<HTMLAnchorElement>,
	dictionary: Record<'movies' | 'persons' | 'tv' | 'new' | 'collections', string>,
} & GeneralProps;

export default function Navigation({
	className,
	testId,
	onClick,
	dictionary,
	...props
}: Props) {
	const routes = Object.keys(navigationRoutes);
	const activeSegment = useSelectedLayoutSegments().find((segment) => routes.includes(segment));

	return (
		<div
			className={`fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 transition-colors duration-150 ${className}`}
			{...props}
		>
			<nav
				className='z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150'
				data-testid={testId}
			>
				<ul className='flex md:flex-wrap md:gap-2 flex-col md:flex-row'>
					{(Object.keys(navigationRoutes) as Array<keyof NavigationRoutes>).map((route) => (
						<li key={route}>
							<Link
								className={`inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 ${activeSegment === route ? 'before:opacity-100' : 'before:opacity-0'}`}
								href={navigationRoutes[route]}
								onClick={onClick}
							>
								{dictionary[route]}
							</Link>
						</li>
					))}
				</ul>
			</nav>
		</div>
	);
}