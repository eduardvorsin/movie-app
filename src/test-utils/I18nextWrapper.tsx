import React, { ReactNode } from 'react';
import i18next from '@/i18n/client';
import { I18nextProvider } from 'react-i18next';

type Props = {
	children: ReactNode,
}

const I18nextWrapper = ({
	children,
}: Props) => {
	return (
		<I18nextProvider i18n={i18next}>
			{children}
		</I18nextProvider>
	);
};

export default I18nextWrapper;