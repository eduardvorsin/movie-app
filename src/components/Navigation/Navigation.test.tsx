import { render, screen } from "@testing-library/react";
import Navigation from "./Navigation";
import i18next from "@/i18n/client";
import I18nextWrapper from "@/test-utils/I18nextWrapper";
import userEvent from "@testing-library/user-event";
import { MouseEventHandler } from "react";

const mockUseSelectedLayoutSegments = jest.fn();
jest.mock('next/navigation', () => ({
	useSelectedLayoutSegments: () => mockUseSelectedLayoutSegments(),
}));

describe('Navigation tests', () => {
	it('is rendered correctly', () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: 'movies',
			persons: 'persons',
			tv: 'series and shows',
			new: 'new releases',
			collections: 'collections',
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-navigation')).toBeInTheDocument();
	});

	it('when clicking on the link, the mock function is called', async () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const user = userEvent.setup();
		const mockFn = jest.fn();
		const clickHandler: MouseEventHandler<HTMLAnchorElement> = (e): void => {
			e.preventDefault();
			mockFn();
		};
		const dictionary = {
			movies: 'movies',
			persons: 'persons',
			tv: 'series and shows',
			new: 'new releases',
			collections: 'collections',
		};
		render(
			<Navigation
				onClick={clickHandler}
				dictionary={dictionary}
				testId='test-navigation'
			/>
		);

		await user.click(screen.getAllByRole<HTMLAnchorElement>('link')[0]);

		expect(mockFn).toHaveBeenCalledTimes(1);
	});

	it('is a basic snapshot', () => {
		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: 'movies',
			persons: 'persons',
			tv: 'series and shows',
			new: 'new releases',
			collections: 'collections',
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>
		);

		expect(screen.getByTestId<HTMLDivElement>('test-navigation')).toMatchSnapshot();
	});
});

describe('Navigation integration tests', () => {
	it('localization into English works correctly', async () => {
		await i18next.changeLanguage('en');
		const translation = new RegExp(i18next.t('navigation.movies'));

		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: i18next.t('navigation.movies'),
			persons: i18next.t('navigation.persons'),
			tv: i18next.t('navigation.tv'),
			new: i18next.t('navigation.new'),
			collections: i18next.t('navigation.collections'),
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLAnchorElement>(translation)).toBeInTheDocument();
	});

	it('localization into Russian works correctly', async () => {
		await i18next.changeLanguage('ru');
		const translation = new RegExp(i18next.t('navigation.movies'));

		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: i18next.t('navigation.movies'),
			persons: i18next.t('navigation.persons'),
			tv: i18next.t('navigation.tv'),
			new: i18next.t('navigation.new'),
			collections: i18next.t('navigation.collections'),
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByText<HTMLAnchorElement>(translation)).toBeInTheDocument();
	});

	it('is a snapshot with English localization', async () => {
		await i18next.changeLanguage('en');

		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: i18next.t('navigation.movies'),
			persons: i18next.t('navigation.persons'),
			tv: i18next.t('navigation.tv'),
			new: i18next.t('navigation.new'),
			collections: i18next.t('navigation.collections'),
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-navigation')).toMatchSnapshot();
	});

	it('is a snapshot with Russian localization', async () => {
		await i18next.changeLanguage('ru');

		mockUseSelectedLayoutSegments.mockReturnValue(['movies', '9999']);
		const mockFn = jest.fn();
		const dictionary = {
			movies: i18next.t('navigation.movies'),
			persons: i18next.t('navigation.persons'),
			tv: i18next.t('navigation.tv'),
			new: i18next.t('navigation.new'),
			collections: i18next.t('navigation.collections'),
		};
		render(
			<Navigation
				onClick={mockFn}
				dictionary={dictionary}
				testId='test-navigation'
			/>,
			{ wrapper: I18nextWrapper }
		);

		expect(screen.getByTestId<HTMLDivElement>('test-navigation')).toMatchSnapshot();
	});
});