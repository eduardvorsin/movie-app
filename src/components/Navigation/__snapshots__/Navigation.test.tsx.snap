// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation integration tests is a snapshot with English localization 1`] = `
<div
  class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300  "
  data-testid="test-navigation"
>
  <nav
    class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
  >
    <ul
      class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
    >
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
          href="/movies"
        >
          movies
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/persons"
        >
          persons
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/tv"
        >
          series and shows
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/new"
        >
          new releases
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/collections"
        >
          collections
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navigation integration tests is a snapshot with Russian localization 1`] = `
<div
  class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300  "
  data-testid="test-navigation"
>
  <nav
    class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
  >
    <ul
      class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
    >
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
          href="/movies"
        >
          фильмы
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/persons"
        >
          люди
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/tv"
        >
          сериалы и шоу
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/new"
        >
          новинки
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/collections"
        >
          коллекции
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`Navigation tests is a basic snapshot 1`] = `
<div
  class="fixed inset-0 bg-neutral-100/90 dark:bg-dark-neutral-100/90 z-0 md:static md:inset-auto md:bg-transparent dark:md:bg-transparent mt-[3.75rem] md:mt-0 md:translate-x-0 transition-colors transition-transform duration-300  "
  data-testid="test-navigation"
>
  <nav
    class="z-100 w-full p-6 pt-3 md:p-0 bg-neutral-200 dark:bg-dark-neutral-250 md:bg-transparent dark:md:bg-transparent flex flex-col md:flex-row transition-colors duration-150"
  >
    <ul
      class="flex md:flex-wrap md:gap-2 flex-col md:flex-row"
    >
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-100"
          href="/movies"
        >
          movies
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/persons"
        >
          persons
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/tv"
        >
          series and shows
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/new"
        >
          new releases
        </a>
      </li>
      <li>
        <a
          aria-disabled="false"
          class="no-underline transition-colors duration-150 text-blue-800 hover:text-blue-900 active:text-blue-1000 dark:text-blue-400 dark:hover:text-blue-300 dark:active:text-blue-200 hover:underline  inline-flex text-200 font-medium py-3 px-1 uppercase md:lowercase w-full md:w-auto md:p-0 relative before:pointer-events-none before:absolute before:bg-current before:rounded-0.5 before:top-1/2 before:-translate-y-1/2 before:h-1/2 before:w-0.5 before:-left-2 md:before:-bottom-1 md:before:left-0 md:before:w-full md:before:top-auto md:before:translate-y-0 md:before:h-0.5 before:transition-opacity before:duration-150 before:opacity-0"
          href="/collections"
        >
          collections
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
